### PR TITLE
BTRX-215 DAC can’t submit a vote after a DUL Reviewed election is re-opened

### PR DESCRIPTION
**Pull request by vvicario**
- [X] **Submitter**: Include the JIRA issue number in the PR description
- [X] **Submitter**: Make sure Swagger is updated if API changes
- [X] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [X] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by ahaessly** [comment]
I'm getting confused between electionId and voteId. 
I thought that a consent has an election with a specific id and version. And that multiple elections for a consent would each have the same electionId but a different version. So what is a voteId? does each vote by a DAC member have a different Id?

File: consent-ws/src/main/java/org/broadinstitute/consent/http/service/ElectionAPI.java [hunk 1]
```diff
@@ -28,6 +28,8 @@
 
     Election describeElectionById(Integer electionId);
 
+    Election describeElectionByVoteId(Integer electionId) throws NotFoundException;
```
Comment (ahaessly) [inline consent-ws/src/main/java/org/broadinstitute/consent/http/service/ElectionAPI.java]: i think this param should be called voteId

**Comment by rushtong** [comment]
@ahaessly 
> So what is a voteId? does each vote by a DAC member have a different Id?

Yes, that is exactly correct. Each DAC member votes in a specific election, so they have a single vote id that records that info.

**Comment by wloforteBroad** [comment]
@ahaessly 
> And that multiple elections for a consent would each have the same electionId but a different version

Each election for the same consent has it's unique id and version number

File: consent-ws/src/main/java/org/broadinstitute/consent/http/service/ElectionAPI.java (hunk 1 above)
Comment (vvicario) [reply consent-ws/src/main/java/org/broadinstitute/consent/http/service/ElectionAPI.java]: @ahaessly you are right, I fixed it! Thanks

File: consent-ws/src/test/java/org/broadinstitute/consent/http/ConsentManageTest.java [hunk 2]
```diff
@@ -49,15 +49,30 @@ public void testConsentManage() throws IOException {
...
         delete(client, electionConsentPathById(CONSENT_ID_2, electionId_2));
     }
 
+    @Test
+    public void testFindElectionByNonExistentVoteId()  throws IOException {
+        Client client = ClientBuilder.newClient();
+        checkStatus(NOT_FOUND, getJson(client, electionVotePathById(999)));
+    }
+
+    private void testFindElectionByVoteId(Vote vote) throws IOException {
```
Comment (ahaessly) [inline consent-ws/src/test/java/org/broadinstitute/consent/http/ConsentManageTest.java]: one nit: it actually takes a vote object and not an Id object. maybe name the method testFindElectionByVote

**Review by ahaessly — APPROVED** [review]
👍  one nit pick. you can decide if you want to change or not.

File: consent-ws/src/test/java/org/broadinstitute/consent/http/ConsentManageTest.java (hunk 2 above)
Comment (vvicario) [reply consent-ws/src/test/java/org/broadinstitute/consent/http/ConsentManageTest.java]: :+1:

**Pull request merged**